test/runtests.jl

```diff
@@ -263,7 +263,7 @@ end
         push!(bunches, sum(table.float_array[start:stop]))
     end
     testvalues = Vector{Float64}([528882.3f0, 1.5877059f6, 2.6465295f6, 3.705353f6, 4.764177f6, 5.823f6, 6.881823f6, 7.9406475f6, 8.999469f6, 1.0058294f7])
-    @test bunches == testvalues
+    @test bunches ≈ testvalues
```

Comment:
isn't the problem here you have `Float32` but your `bunches` are Float64?

Reply:
Btw. I can reproduce the error on my machine. Johannes is currently investigating why he can't. Anyways, good catch with the Float32 stuff!

Reply:
Ok I'm kinda confused ... so what I do is:

1. Making sure I'm on the right commit, which is `commit a6719f3ecb8763a307571aec177630d326b339c4` according to `git log -1`
2. Checking my env is clean, which is according to `git status`
3. Switching inside docker env: `docker run -it --rm -v `pwd`:/test julia:1.7 /bin/bash`
4. There I run `julia --project=@. -e"using Pkg; Pkg.instantiate(); Pkg.add([\"MD5\",\"ThreadsX\"]); include(\"test/runtests.jl\")"` inside `/test`

and I get:
```
Test Summary:           | Pass  Total
LazyBranch and LazyTree |   26     26
```
:confused:

Reply:
Whatever I do, I cannot reproduce, I always get 

```
LazyBranch and LazyTree: Test Failed at /Users/tamasgal/Dev/UnROOT.jl/test/runtests.jl:266
  Expression: bunches ≈ testvalues
   Evaluated: [528882.25, 1.587705875e6, 2.6465295e6, 3.705353e6, 4.764177e6, 5.8230005e6, 6.8818235e6, 7.9406475e6, 8.999469e6, 1.0058295e7] ≈ [528882.3125, 1.587705875e6, 2.6465295e6, 3.705353e6, 4.764177e6, 5.823e6, 6.881823e6, 7.9406475e6, 8.999469e6, 1.0058294e7]
```

But something with your "Docker-workflow" is not working either. The way you described it gives me other errors with Julia 1.7.

Do you have something weird in your startup files? As @Moelf pointed out, the `Float32` stuff is also a bit fishy, but that does not explain why it works for you. Maybe it's a Julia 1.7 thing? 

Anyways, I'd propose the good old `juila --project=.` (inside the top folder) and then `]test`. I use that all the time, everywhere 😉

Reply:
Regarding my workflow maybe it was a copy&paste thing that the \` arround pwd were not displayed correctly, so I changed to `$PWD`. Yeah running with `]test` reproduces the error, but I still find I weird that running `julia --project=@. -e"using Pkg; Pkg.instantiate(); Pkg.add([\"MD5\",\"ThreadsX\"]); include(\"test/runtests.jl\")"` which basically should do the same just lets the tests pass. Regarding something weird in the startup files: my home dir is not mounted to the docker and repo is clean. So there should not be something messy stuff going on in the background.

Reply:
I am not sure but it might be that `MD5` and `ThreadsX` do something weird with the dependencies -- unexpected downgrades or whatever.

Reply:
`julia --project=@.` is wrong

Reply:
Ah, I did not see the @ 😆

Reply:
Me neither, just reused from command history🤦🏻‍♂️

Reply:
well can you print `Pkg.status()` after `--project=.` and show us `git diff` or whatever?

Reply:
The `git diff` is empty and `julia --project=. -e"using Pkg; Pkg.status();"` shows:
```
Project UnROOT v0.10.6
      Status `/test/Project.toml`
  [1520ce14] AbstractTrees v0.4.4
  [65a8f2f4] ArraysOfArrays v0.6.1
  [c3b6d118] BitIntegers v0.2.7
  [5ba52731] CodecLz4 v0.4.0
  [ba30903b] CodecXz v0.7.0
  [6b39b394] CodecZstd v0.7.2
  [cd3eb016] HTTP v1.7.4
  [c8e1da08] IterTools v1.4.0
  [8ac3fa9e] LRUCache v1.4.1
  [9255714d] LibDeflate v0.4.1
  [3f54b04b] LorentzVectors v0.4.0
  [6fafb56a] Memoization v0.2.0
  [2a8e4939] Mixers v0.1.2
  [d96e819e] Parameters v0.12.3
  [08abe8d2] PrettyTables v2.2.3
  [91c51154] SentinelArrays v1.3.18
  [90137ffa] StaticArrays v1.5.21
  [09ab397b] StructArrays v0.6.15
  [bd369af6] Tables v1.10.1
  [9d84c17e] xrootdgo_jll v0.32.1+0
  [a63ad114] Mmap
```

Reply:
??? why are you activating under `/test` folder?

Reply:
I added a `@test false` in order to make sure that this script is executed which worked like expected. So the behavior was like before: one method failed at the array comparison and the other method at the `@test false`. Anyways: I mixed something up, the type of the branch is `Float32` so the `sum` is `Float32` and written to the `Vector{Float64}` ... so I changed it at every spot to the correct type. Sorry for all the confusion :sweat_smile:

Reply:
>I added a @test false in order to make sure that this script is executed which worked like expected.

I'm not saying you're not executing the script `runtest.jl`, but you're using the UnROOT from your home environment.

The fact that `status()` doesn't show UnROOT.jl is worrisome -- what `UnROOT.jl` are you actually using?